benchmarks/performancepoetry/src/main/java/com/squareup/benchmarks/performance/poetry/PerformancePoemWorkflow.kt

```diff
@@ -0,0 +1,244 @@
+package com.squareup.benchmarks.performance.poetry
+
+import com.squareup.benchmarks.performance.poetry.PerformancePoemWorkflow.Action.ClearSelection
+import com.squareup.benchmarks.performance.poetry.PerformancePoemWorkflow.Action.HandleStanzaListOutput
+import com.squareup.benchmarks.performance.poetry.PerformancePoemWorkflow.Action.SelectNext
+import com.squareup.benchmarks.performance.poetry.PerformancePoemWorkflow.Action.SelectPrevious
+import com.squareup.benchmarks.performance.poetry.PerformancePoemWorkflow.State
+import com.squareup.benchmarks.performance.poetry.PerformancePoemWorkflow.State.ComplexCall
+import com.squareup.benchmarks.performance.poetry.PerformancePoemWorkflow.State.Initializing
+import com.squareup.benchmarks.performance.poetry.PerformancePoemWorkflow.State.Selected
+import com.squareup.benchmarks.performance.poetry.views.BlankScreen
+import com.squareup.sample.container.overviewdetail.OverviewDetailContainer
+import com.squareup.sample.container.overviewdetail.OverviewDetailScreen
+import com.squareup.sample.poetry.PoemWorkflow
+import com.squareup.sample.poetry.PoemWorkflow.ClosePoem
+import com.squareup.sample.poetry.SimulatedPerfConfig
+import com.squareup.sample.poetry.StanzaListWorkflow
+import com.squareup.sample.poetry.StanzaListWorkflow.NO_SELECTED_STANZA
+import com.squareup.sample.poetry.StanzaRendering
+import com.squareup.sample.poetry.StanzaWorkflow
+import com.squareup.sample.poetry.StanzaWorkflow.Output.CloseStanzas
+import com.squareup.sample.poetry.StanzaWorkflow.Output.ShowNextStanza
+import com.squareup.sample.poetry.StanzaWorkflow.Output.ShowPreviousStanza
+import com.squareup.sample.poetry.StanzaWorkflow.Props
+import com.squareup.sample.poetry.model.Poem
+import com.squareup.workflow1.Snapshot
+import com.squareup.workflow1.StatefulWorkflow
+import com.squareup.workflow1.Worker
+import com.squareup.workflow1.WorkflowAction
+import com.squareup.workflow1.WorkflowAction.Companion.noAction
+import com.squareup.workflow1.action
+import com.squareup.workflow1.runningWorker
+import com.squareup.workflow1.ui.AndroidViewRendering
+import com.squareup.workflow1.ui.BuilderViewFactory
+import com.squareup.workflow1.ui.ViewFactory
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.backstack.BackStackScreen
+import com.squareup.workflow1.ui.backstack.toBackStackScreen
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+
+/**
+ * Renders a [Poem] as a [OverviewDetailScreen], whose overview is a [StanzaListRendering]
+ * for the poem, and whose detail traverses through [StanzaRendering]s.
+ */
+public class PerformancePoemWorkflow(
+  private val simulatedPerfConfig: SimulatedPerfConfig = SimulatedPerfConfig.NO_SIMULATED_PERF,
+  private val isLoading: MutableStateFlow<Boolean>
```

Comment:
This would be more than a little sketchy in real life, maybe document what you're up to. I assume `isLoading` can be toggled by both this workflow and its parent?

Reply:
yes it can. Perhaps I need to step back though if this is smelly to you. I thought this was your suggestion wrt how Gatekeepers would overlay all children via an injected signal?

Reply:
I can extend `OverviewDetailScreen` to an `MaybeLoadingOverviewDetailScreen` and handle it that way if we think this is preferable and better practice?

Reply:
I'm just saying that it's sketchy to have `MutableStateFlow` floating around as public API rather than an implementation detail. I'd expect the lower level workflows to be injecting an interface that lets them request the loading UI be shown or hidden, but that doesn't give them access to the stream of loading values.

In real life, we'd tie that to a token in case of overlapping show/hide requests. Probably overkill for this, but just for the record:

```kotlin
interface Loader {
  fun startLoading(token: String)
  fun stopLoading(token: String)
}
```

```kotlin
class RealLoader: Loader {
  private val loaders = mutableSetOf<String>()
  private val _showIt = MutableStateFlow(false)

  public val showIt(): StateFlow<Boolean> = _showIt

  override fun startLoading(token: String) {
    loaders.add(token)
    _showIt.value = true
  }


  override fun stopLoading(token: String) {
    loaders.remove(token)
    _showIt.value = loaders.isNotEmpty()
  }  
}
```

Alos, I don't think we'd want to make `OverviewDetailScreen` non-final.

Reply:
Added the note to make that clear and hint at this type of a solution.